ext/include/opentelemetry/ext/http/client/http_client_factory.h

```diff
@@ -0,0 +1,34 @@
+#pragma once
+#include "opentelemetry/ext/http/client/http_client.h"
+#ifdef WITH_CURL
+#  include "opentelemetry/ext/http/client/curl/http_client_curl.h"
+#endif
+
+OPENTELEMETRY_BEGIN_NAMESPACE
+namespace ext
+{
+namespace http
+{
+namespace client
+{
+
+class HttpClientFactory
+{
+public:
+#ifdef WITH_CURL
+  static std::shared_ptr<SessionManager> Create()
+  {
+
+    return std::make_shared<ext::http::client::curl::SessionManager>();
```

Comment:
Just wrap the `return` statement in `#if`?

Reply:
Yes can be done. This looks bit more cleaner in vscode, as it would show the specific `Create()` function highlighted based on available client library. But was my personal choice, and I can change if it looks ugly to read.

Reply:
I have strong preference on this. I was thinking about that unifying the method signature indicates that it is the same function with different body instead of different functions for CURL and else.

Reply:
sure will change it if that's causing confusion.

Reply:
Done.